Dockerfile

```diff
@@ -15,7 +15,7 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
-ARG GO_VERSION=1.21.5
+ARG GO_VERSION=1.21.6
 ARG XX_VERSION=1.2.1
 ARG GOLANGCI_LINT_VERSION=v1.54.2
```

Comment:
Not for this PR, but there's newer versions of xx (v1.3.0) and GolangCI-lint (v1.55.2) that we may want to update as well.

For xx; I'm running into an  issue when cross-compiling in https://github.com/docker/docker-credential-helpers/pull/310, so just a heads-up on that.

Reply:
Thanks for the heads up, I'll open a couple new PRs for these

Reply:
golangci-lint PR @ #11332 

Following the `xx` discussion, same issue on Compose side:
```
 > [linux/arm64->darwin/amd64 build 1/1] RUN --mount=type=bind,target=.     --mount=type=cache,target=/root/.cache     --mount=type=cache,target=/go/pkg/mod     --mount=type=bind,from=osxcross,src=/osxsdk,target=/xx-sdk     xx-go --wrap &&     if [ "$(xx-info os)" == "darwin" ]; then export CGO_ENABLED=1; fi &&     make build GO_BUILDTAGS="e2e" DESTDIR=/out &&     xx-verify --static /out/docker-compose:
1.355 GO111MODULE=on go build  -trimpath -tags "e2e" -ldflags "-w -X github.com/docker/compose/v2/internal.Version=v2.24.0-birthday.10-39-gf5c53c2d0.m" -o "/out/docker-compose" ./cmd
201.4 # github.com/docker/compose/v2/cmd
201.4 /usr/local/go/pkg/tool/linux_arm64/link: running x86_64-apple-macos10.6-clang failed: exit status 1
201.4 /usr/bin/x86_64-apple-macos10.6-ld: line 9: sigtool-gensig: not found
201.4 clang: error: unable to execute command: No such file or directory
201.4 clang: error: linker command failed due to signal (use -v to see invocation)
201.4
```

Reply:
Thanks for checking! /cc @crazy-max